### PR TITLE
feat: update the visualization of traces now that runnables are not produced anymore

### PR DESCRIPTION
**Pull request by etorreborre**
The previous animation of traces was relying on the fact that a list of runnables to execute was included in the trace file.
This PR now keeps track of that list of runnables based on input and external effects.

I also fixed the display of stage circles in Firefox and Safari.

## Summary by CodeRabbit

* **New Features**
  * Incremental, deduplicated management of the runnable stack for more predictable animation state updates.
  * Node sizing now initialized during rendering to ensure consistent visual setup across environments.

* **Bug Fixes**
  * More reliable handling of resume, input, and suspend events to keep animation stages consistent.
  * Consistent rendering of animation nodes for stable playback and fewer edge-case glitches.

### CODE REVIEW

File: simulation/amaru-sim/tests/animations/traces.html [hunk 1]
```diff
@@ -1001,6 +1001,58 @@ <h1>Traces animation</h1>
...
+        const stageName = displayStageName(name);
+        if (!Array.isArray(currentRunnable)) {
+            currentRunnable = [];
+        }
+        // Only add if not already present
+        const exists = currentRunnable.some(it => it && it.name === stageName);
+        if (exists) return;
+        currentRunnable = currentRunnable.concat([{name: stageName}]);
+        updateStack();
+    }
```
[the remark this answers is not in the document]
Comment (etorreborre) [reply simulation/amaru-sim/tests/animations/traces.html:1033]: That's confusing but correct. Some effects have both a `type` and `effect_type` field.

File: simulation/amaru-sim/tests/animations/traces.html [hunk 2]
```diff
@@ -1001,6 +1001,58 @@ <h1>Traces animation</h1>
...
+            name = eff.at_stage;
+        } else {
+            return;
+        }
+        const stageName = displayStageName(name);
+        if (!Array.isArray(currentRunnable)) {
+            currentRunnable = [];
+        }
+        // Only add if not already present
+        const exists = currentRunnable.some(it => it && it.name === stageName);
```
Comment (abailly) [inline simulation/amaru-sim/tests/animations/traces.html:1029]: There seems to be some opportunity for reducing duplication here 🤔

Comment (etorreborre) [reply simulation/amaru-sim/tests/animations/traces.html:1029]: Done.
